x-pack/plugins/apm/public/utils/formatters.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { memoize } from 'lodash';
+
+// tslint:disable-next-line  no-var-requires
+const numeral: (input: number) => Numeral = require('@elastic/numeral');
+
+interface Numeral {
+  format: (pattern: string) => string;
+}
+
+const UNIT_CUT_OFF = 10 * 1000000; // 10 seconds in microseconds
+
+export function asSeconds(value: number, withUnit: boolean = true) {
+  const formatted = asDecimal(value / 1000000);
+  return `${formatted}${withUnit ? ' s' : ''}`;
+}
+
+export function asMillis(value: number, withUnit: boolean = true) {
+  const formatted = asInteger(value / 1000);
+  return `${formatted}${withUnit ? ' ms' : ''}`;
+}
+
+export function asMillisWithDefault(value?: number) {
+  if (value == null) {
+    return `N/A`;
+  }
+  return asMillis(value);
+}
+
+export const getTimeFormatter: (
+  max: any
```

Comment:
```suggestion
  max: number
```

Reply:
the issue with number is that it's being passed the result of `d3.max(buckets)` in `x-pack/plugins/apm/public/components/app/TransactionDetails/Distribution/index.tsx:69` which returns type `string | undefined`, not sure if that was intentional or not. but the conclusion i reach here is to define `any` type for now until we look into that call.

Reply:
Damn, you are right. We've got an error in our types. Good catch! 

`IChartPoint` in `app/TransactionDetails/Distribution/index.tsx` should be:
```ts
interface IChartPoint {
  sample?: IBucket['sample'];
  x0: number;
  x: number;
  y: number;
  style: {
    cursor: string;
  };
}
```

and `IChartPoint` should be added as return value for `getFormattedBuckets` to enforce the relationship between `IChartPoint` and `IBucket`

`IBucket` in `lib/transactions/distribution/get_buckets.ts` should be changed to:
```ts
export interface IBucket {
  key: number;
  count: number;
  sample?: IBucketSample;
}
```

Can I ask you to fix it as part of this PR? We are still in the humble beginnings of migrating to TS, so we'll stumble upon incorrect types now and then. I hope we can fix it when we see it (unless it's too big a change) so we improve on what we have, and don't lose trust in the types (i feel I need a "don't stop believing" gif here)